test/features/test_organized_edge_detection.cpp

```diff
@@ -0,0 +1,125 @@
+  const float SYNTHETIC_CLOUD_DEPTH_DISCONTINUITY = .02f;
+  const float SYNTHETIC_CLOUD_RESOLUTION = 0.01f;
+
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud_;
+  pcl::PointIndicesPtr indices_;
+
+  void
+  SetUp() override
+  {
+    cloud_ = generateSyntheticEdgeDetectionCloud();
+  }
+
+private:
+  pcl::PointCloud<pcl::PointXYZ>::Ptr
+  generateSyntheticEdgeDetectionCloud()
+  {
+    auto organized_test_cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>>(
+        OUTER_SQUARE_EDGE_LENGTH, OUTER_SQUARE_EDGE_LENGTH);
+
+    // Draw a smaller square in front of a larger square both centered on the
+    // view axis to generate synthetic occluding and occluded edges based on depth
+    // discontinuity between neighboring pixels.  The base depth and resolution are
+    // arbitrary and useful for visualizing the cloud.  The discontinuity of the
+    // generated cloud must be greater than the threshold set when running the
+    // organized edge detection algorithm.
+    for (auto row = 0; row < OUTER_SQUARE_EDGE_LENGTH; ++row) {
+      for (auto col = 0; col < OUTER_SQUARE_EDGE_LENGTH; ++col) {
+        float x = col - (OUTER_SQUARE_EDGE_LENGTH / 2);
+        float y = row - (OUTER_SQUARE_EDGE_LENGTH / 2);
+
+        const auto outer_square_ctr = OUTER_SQUARE_EDGE_LENGTH / 2;
+        const auto inner_square_ctr = INNER_SQUARE_EDGE_LENGTH / 2;
+
+        auto depth = SYNTHETIC_CLOUD_BASE_DEPTH;
+
+        // If pixels correspond to smaller box, then set depth and color appropriately
+        if (col >= outer_square_ctr - inner_square_ctr &&
+            col < outer_square_ctr + inner_square_ctr) {
+          if (row >= outer_square_ctr - inner_square_ctr &&
+              row < outer_square_ctr + inner_square_ctr) {
+            depth = SYNTHETIC_CLOUD_BASE_DEPTH - SYNTHETIC_CLOUD_DEPTH_DISCONTINUITY;
+          }
+        }
+
+        organized_test_cloud->at(col, row).x = x * SYNTHETIC_CLOUD_RESOLUTION;
+        organized_test_cloud->at(col, row).y = y * SYNTHETIC_CLOUD_RESOLUTION;
+        organized_test_cloud->at(col, row).z = depth;
+      }
+    }
+
+    return organized_test_cloud;
+  }
+};
+} // namespace
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*
+This test is designed to ensure that the organized edge detection appropriately
+classifies occluding and occluding edges and to specifically detect the type of
+regression detailed in PR 4275 (https://github.com/PointCloudLibrary/pcl/pull/4275).
+This test works by generating a synthetic cloud of one square slightly in front of
+another square, so that occluding edges and occluded edges are generated.  The
+regression introduced in PCL 1.10.1 was a logic bug that caused both occluding and
+occluded edges to be miscategorized as occluding edges.  This test should catch
+this and similar bugs.
+*/
+TEST_F(OrganizedPlaneDetectionTestFixture, OccludedAndOccludingEdges)
+{
+  const auto MAX_SEARCH_NEIGHBORS = 50;
```

Comment:
Should we not tighten this to a neighborhood of 8, to just look at adjacent pixels in the immediate vicinity?